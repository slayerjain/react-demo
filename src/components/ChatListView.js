import React from 'react';
import { ChatList } from 'react-chat-elements';


const ChatListView = () => (
  <div className="content-container--chat-list">
    <ChatList
      className="chat-list content-container--chat-list-item"
      dataSource={[
        {
          avatar: 'https://facebook.github.io/react/logo-og.png',
          alt: 'Reactjs',
          title: 'Shubham',
          subtitle: 'What are you doing?',
          date: new Date(),
          unread: 0,
        },
        {
          alt: 'Reactjs',
          title: 'Pankaj',
          subtitle: 'Here are my Powers',
          date: new Date(),
          unread: 2,
        },
        ]}
    />
  </div>
);

export default ChatListView;
