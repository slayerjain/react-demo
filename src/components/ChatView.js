import React from 'react';
import { MessageList, Input, Button } from 'react-chat-elements';
const ChatView = () => (
  <div className="content-container--message-list">
    <MessageList
      className="message-list"
      lockable
      toBottomHeight="100%"
      dataSource={[
        {
          position: 'right',
          type: 'text',
          text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
          date: new Date(),
        },
        {
          position: 'left',
          type: 'text',
          text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
          date: new Date(),
        },
        ]}
    />
    <Input
      className="content-container--message-input"
      placeholder="Type here..."
      autoHeight
      multiline
      rightButtons={
        <Button
          color="white"
          text="Send"
        />
      }
    />
  </div>
);

export default ChatView;
