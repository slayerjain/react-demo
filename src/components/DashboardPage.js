import React from 'react';
import { SideBar } from 'react-chat-elements';

const DashboardPage = () => (
  <SideBar
    top={
      <div>'TOP' area</div>
    }
    center={
      <div>'CENTER' area</div>
    }
    bottom={
      <div>'BOTTOM' area</div>
    }
  />
);

export default DashboardPage;
