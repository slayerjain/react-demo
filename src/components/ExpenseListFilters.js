import React from 'react';
import uuid from 'uuid'
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';


import { setEndDate, setStartDate, setTextFilter, sortByAmount, sortByDate } from '../actions/filters'

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null,
  }

  onDatesChanged = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChanged = calendarFocused => {
    this.setState(() => ({ calendarFocused }));
  };

  onTextChange = e => {
    this.props.setTextFilter(e.target.value);
  }

  onSortChange = e => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  }

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId={uuid()}
          endDate={this.props.filters.endDate}
          endDateId={uuid()}
          onDatesChange={this.onDatesChanged}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChanged}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => (false)}
        />
      </div>
    )
  }
}

const mapStateToProps = state => ({
  filters: state.filters,
});

const mapDispatchToProps = dispatch => ({
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: startDate => dispatch(setStartDate(startDate)),
  setEndDAte: endDate => dispatch(setEndDate(endDate)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
