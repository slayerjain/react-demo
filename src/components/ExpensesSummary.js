import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import totalExpenses from '../selectors/total-expenses';

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => (
  <div>
    <h1>
    {
      `Viewing ${expenseCount} expense${expenseCount > 1 ? 's' : ''} totaling ${numeral(expenseTotal/ 100).format('$0,0.00')}`
    }
    </h1>
  </div>
);

const mapStateToProps = state => {
  const selectedExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: selectedExpenses.length,
    expenseTotal: totalExpenses(selectedExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
