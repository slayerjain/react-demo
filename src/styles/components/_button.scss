.button {
  padding: $s-size;
  border: none;
  background-color: $blue;
  color: white;
  display: inline-block;
  font-weight: 300;
  line-height: 1;
  font-size: $font-size-large;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background-color: #8888;
}
