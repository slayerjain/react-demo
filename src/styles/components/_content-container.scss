.content-container {
  margin: 0 auto;
  padding: 0 $m-size;
  max-width: 110rem;
}

.content-container--chat-view-split {
  display: flex;
  flex-direction: row;
  //justify-content: space-between;
  /* But in order to take all the available space we need
     to use this */
  //position: absolute;
  //top: 0;
  padding: 0;
  height: calc(100vh - 60px);
  //bottom: 0;
  overflow: auto;
}

.content-container--chat-list {
  margin: 0;
  width: 35%;
}

.content-container--chat-list-item {
  width: 100%;
}

.content-container--message-list {
  width: 100%;
  background-color: #eee;
  display: flex;
  flex-direction: column;
}

.content-container--message-input {

}