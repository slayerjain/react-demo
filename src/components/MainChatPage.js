import React from 'react';
import ChatListView from './ChatListView';
import ChatView from './ChatView';

const MainChatPage = () => (
  <div className="content-container content-container--chat-view-split">
    <ChatListView />
    <ChatView />
  </div>
);

export default MainChatPage;
