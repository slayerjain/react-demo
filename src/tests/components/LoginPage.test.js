import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should render loginPage correctly', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  const spy = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={spy} />);
  wrapper.find('button').simulate('click');
  expect(spy).toHaveBeenCalled();
});
