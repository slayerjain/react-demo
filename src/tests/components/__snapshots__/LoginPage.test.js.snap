// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render loginPage correctly 1`] = `
<div>
  <h3>
    Login Page
  </h3>
  <button>
    Login
  </button>
</div>
`;
